packages/mdc-shape/component.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDCComponent} from '@material/base';
+import MDCShapeFoundation from './foundation';
+import {getElevationFromDOM, getBackgroundFromDOM} from './util';
+
+export default class MDCShape extends MDCComponent {
+  set background(value) {
+    this.foundation_.setBackground(value);
+  }
+
+  set elevation(value) {
+    this.foundation_.setElevation(value, true);
+  }
+
+  redraw() {
+    this.foundation_.redraw();
+  }
+
+  createAdapter() {
+    return {setCanvasWidth: (value) => this.canvas_.width = value,
+      setCanvasHeight: (value) => this.canvas_.height = value,
+      getCanvasWidth: () => this.canvas_.offsetWidth,
+      getCanvasHeight: () => this.canvas_.offsetHeight,
+      getDevicePixelRatio: () => window.devicePixelRatio,
+      create2dRenderingContext: () => this.canvas_.getContext('2d'),
+      createPath2D: (path) => new Path2D(path),
+      setClipPathData: (value) => this.path_.setAttribute('d', value)};
+  }
+
+  initialSyncWithDOM() {
+    this.foundation_.setBackground(getBackgroundFromDOM(this.root_), false);
+    this.foundation_.setElevation(getElevationFromDOM(this.root_), false);
```

Comment:
AFAICT, `getElevationFromDOM` assumes that CSS custom properties work, which in turn makes me suspect we need to be feature-detecting this the same way that we're doing in ripple (and IIRC a couple of other places, that probably need updating)...

Maybe it's time for #1104?

What's our plan for browsers that don't support custom CSS properties?

OTOH, is there a reason that setting elevation needs to be done this way rather than actually reusing mdc-elevation classes (esp. if there's any desire to support no-JS shapes), or just using a data attribute if this is solely for the purpose of initialization via JS?

Reply:
Theres no support for no-JS shapes...Shape JS has to draw the shape on a canvas, and it has to draw all the shadows on the canvas. The existing Sass mixins for elevation won't work with shapes because the shadows won't properly follow the irregular edges of a shape.